client/components/workout-detail.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { format } from 'date-fns';
+import Header from './header';
+import JournalPage from '../pages/journal';
+import { exercises, sets } from '../lib/getWorkoutDetails';
+
+class WorkoutDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      exercises: [],
+      date: ''
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  formatDate(date) {
+    const newDate = new Date(date);
+    const formattedDate = format(newDate, 'MMMM dd, yyyy');
+    return formattedDate;
+  }
+
+  componentDidMount() {
+    fetch(`/api/sets/${this.props.workoutId} `)
+      .then(res => res.json())
+      .then(result => this.setState({
+        exercises: sets(exercises(result), result),
+        date: this.formatDate(result[0].workoutDate)
+      }))
```

Comment:
Rather than make the client reformat the data, the server should do it. Think of it like going to a restaurant. It would be sort of a pain if they just gave you the ingredients for your meal instead of actually preparing a plate of food. 😂